File: app/src/main/java/com/example/ase_project_nutrismart/ItemSelectedFragment.java
package com.example.ase_project_nutrismart;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.RecoverySystem;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.example.ase_project_nutrismart.Adapter.PurchasedGroceryAdapter;
import com.example.ase_project_nutrismart.Adapter.SelectedGroceryAdapter;
import com.example.ase_project_nutrismart.Model.SelectedGroceryModel;
import com.example.ase_project_nutrismart.Response.Grocery;
import com.example.ase_project_nutrismart.Response.SelectedGrocery;
import com.example.ase_project_nutrismart.Retrofit.APIClient;
import com.example.ase_project_nutrismart.Retrofit.ApiInterface;
import com.example.ase_project_nutrismart.ui.gallery.GroceryFragment;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * A simple {@link Fragment} subclass.
 * create an instance of this fragment.
 */
public class ItemSelectedFragment extends Fragment {
    ApiInterface apiInterface;
RecyclerView RvSelectedItems;
    ArrayList<Grocery>sendList=new ArrayList<>();
    Button submit;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view= inflater.inflate(R.layout.fragment_item_selected, container, false);

        apiInterface = APIClient.getClient().create(ApiInterface.class);
        RvSelectedItems=view.findViewById(R.id.RvSelectedItems);
        submit=view.findViewById(R.id.submit);
        getPurchasedGroceryApi();

submit.setOnClickListener(new View.OnClickListener() {
    @Override
    public void onClick(View view) {
        sendData();
    }
});
        // Inflate the layout for this fragment
        return view;
    }

    private void sendData() {

        SelectedGroceryModel model=new SelectedGroceryModel();
        model.setData(sendList);
        model.purchasedDate="2022-11-13";
        model.userEmail="sd@yopmail.com";
        Call<SelectedGrocery> send = apiInterface.sendSelectedGrocery(model);
        send.enqueue(new Callback<SelectedGrocery>() {
            @Override
            public void onResponse(Call<SelectedGrocery> call, Response<SelectedGrocery> response) {

                Log.d("dkfkfg",""+response.code());
                if(response.code()==201)
                {
                    NavController navController = Navigation.findNavController(getActivity(), R.id.nav_host_fragment_content_home_actvity);
                    navController.navigate(R.id.tracker);
                }
            }

            @Override
            public void onFailure(Call<SelectedGrocery> call, Throwable t) {

            }
        });


    }

    private void getPurchasedGroceryApi() {


        SelectedGroceryModel model=new SelectedGroceryModel();
        ArrayList <Grocery>list=new ArrayList();
        model.setData(list);
        model.purchasedDate="2022-11-13";
        model.userEmail="sd@yopmail.com";
        Call<SelectedGrocery> send = apiInterface.sendSelectedGrocery(model);
        send.enqueue(new Callback<SelectedGrocery>() {
            @Override
            public void onResponse(Call<SelectedGrocery> call, Response<SelectedGrocery> response) {

                Log.d("dkfkfg",""+response.code());
sendList=response.body().getData();
                SelectedGroceryAdapter adapter = new SelectedGroceryAdapter(ItemSelectedFragment.this,response.body().getData());
                // Attach the adapter to the recyclerview to populate items
                RvSelectedItems.setAdapter(adapter);
                // Set layout manager to position the items
                RvSelectedItems.setLayoutManager(new LinearLayoutManager(getContext()));

//                if(response.code()==201)
//                {
//                    NavController navController = Navigation.findNavController(getActivity(), R.id.nav_host_fragment_content_home_actvity);
//                    navController.navigate(R.id.itemSlected);
//                }
            }

            @Override
            public void onFailure(Call<SelectedGrocery> call, Throwable t) {

            }
        });

    }
}

File: app/src/main/java/com/example/ase_project_nutrismart/ItemSelectedFragment.kt
package com.example.ase_project_nutrismart

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.ase_project_nutrismart.Adapter.SelectedGroceryAdapter
import com.example.ase_project_nutrismart.Model.SelectedGroceryModel
import com.example.ase_project_nutrismart.Response.Grocery
import com.example.ase_project_nutrismart.Response.SelectedGrocery
import com.example.ase_project_nutrismart.Retrofit.APIClient.client
import com.example.ase_project_nutrismart.Retrofit.ApiInterface
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * A simple [Fragment] subclass.
 * create an instance of this fragment.
 */
class ItemSelectedFragment : Fragment() {
   lateinit var apiInterface: ApiInterface
    lateinit var  RvSelectedItems: RecyclerView
    lateinit var sendList: ArrayList<Grocery>
    lateinit var submit: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_item_selected, container, false)
        apiInterface = client!!.create(ApiInterface::class.java)
        RvSelectedItems = view.findViewById(R.id.RvSelectedItems)
        submit = view.findViewById(R.id.submit)
        purchasedGroceryApi
        submit.setOnClickListener(View.OnClickListener { sendData() })
        // Inflate the layout for this fragment
        return view
    }

    private fun sendData() {
        val model = SelectedGroceryModel()
        model.data = sendList
        model.purchasedDate = "2022-11-13"
        model.userEmail = "sd@yopmail.com"
        val send = apiInterface!!.sendSelectedGrocery(model)
        send!!.enqueue(object : Callback<SelectedGrocery?> {
            override fun onResponse(
                call: Call<SelectedGrocery?>,
                response: Response<SelectedGrocery?>
            ) {
                Log.d("dkfkfg", "" + response.code())
                if (response.code() == 201) {
                    val navController =
                        findNavController(activity!!, R.id.nav_host_fragment_content_home_actvity)
                    navController.navigate(R.id.tracker)
                }
            }

            override fun onFailure(call: Call<SelectedGrocery?>, t: Throwable) {}
        })
    }
    // Attach the adapter to the recyclerview to populate items
    // Set layout manager to position the items

    //                if(response.code()==201)
//                {
//                    NavController navController = Navigation.findNavController(getActivity(), R.id.nav_host_fragment_content_home_actvity);
//                    navController.navigate(R.id.itemSlected);
//                }
    private val purchasedGroceryApi: Unit
        private get() {
            val model = SelectedGroceryModel()
            val list: ArrayList<Grocery> = ArrayList<Grocery>()
            model.data = list
            model.purchasedDate = "2022-11-13"
            model.userEmail = "sd@yopmail.com"
            val send = apiInterface!!.sendSelectedGrocery(model)
            send!!.enqueue(object : Callback<SelectedGrocery?> {
                override fun onResponse(
                    call: Call<SelectedGrocery?>,
                    response: Response<SelectedGrocery?>
                ) {
                    Log.d("dkfkfg", "" + response.code())
                    sendList = response.body()!!.data!!
                    val adapter =
                        SelectedGroceryAdapter(this@ItemSelectedFragment,sendList)
                    // Attach the adapter to the recyclerview to populate items
                    RvSelectedItems!!.adapter = adapter
                    // Set layout manager to position the items
                    RvSelectedItems!!.layoutManager = LinearLayoutManager(context)

//                if(response.code()==201)
//                {
//                    NavController navController = Navigation.findNavController(getActivity(), R.id.nav_host_fragment_content_home_actvity);
//                    navController.navigate(R.id.itemSlected);
//                }
                }

                override fun onFailure(call: Call<SelectedGrocery?>, t: Throwable) {}
            })
        }
}